### PR TITLE
Improve test coverage for Pool and Staking contracts

### PR DESCRIPTION
**Pull request by francisldn**
#749 Expand test coverage across existing contracts
Added more tests to Pool - unit.js and Staking - unit.js test files to improve coverage to 100%.

### CODE REVIEW

File: source/pool/test/unit.js [hunk 1]
```diff
@@ -1,607 +1,653 @@
...
+    it('set stake contract successful', async () => {
+      await pool.connect(deployer).setStakingContract(stakeContract.address)
+      expect(await pool.stakingContract()).to.equal(stakeContract.address)
+    })
+
+    it('set stake contract reverts', async () => {
+      await expect(
+        pool
+          .connect(deployer)
+          .setStakingContract('0x0000000000000000000000000000000000000000')
```
Comment (AboldUSER) [inline source/pool/test/unit.js:130]: I know this address 0x000... was there from a previous commit, though this is a good opportunity to replace it with a variable from @airswap/constants.
 I suggest importing the zero address constant into the test file, and using the constant in place of the current 0x000...
 `const { ADDRESS_ZERO } = require('@airswap/constants')`
 You can see how this is used in other test files like ./source/swap/test/unit.js

File: source/pool/test/unit.js [hunk 2]
```diff
@@ -1,607 +1,653 @@
...
+      await feeToken2.mock.approve.returns(true)
+      await pool.connect(deployer).setStakingToken(feeToken2.address)
+      expect(await pool.stakingToken()).to.equal(feeToken2.address)
+    })
+
+    it('set stake token reverts', async () => {
+      await expect(
+        pool
+          .connect(deployer)
+          .setStakingToken('0x0000000000000000000000000000000000000000')
```
Comment (AboldUSER) [inline source/pool/test/unit.js:144]: Same as above zero address comment

File: source/pool/test/unit.js [hunk 3]
```diff
@@ -1,607 +1,653 @@
...
+        )
+      ).to.emit(pool, 'Withdraw')
+
+      const isClaimed = await pool.claimed(root, alice.address)
+      expect(isClaimed).to.equal(true)
+    })
+
+    it('withdrawWithRecipient reverts with minimumAmount not met', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
```
Comment (AboldUSER) [inline source/pool/test/unit.js:323]: This was already present from a previous commit, but in review I realize it is not needed, so I suggest removing this line.
`await feeToken.mock.transfer.returns(true)`

File: source/pool/test/unit.js [hunk 4]
```diff
@@ -1,607 +1,653 @@
...
+
+      const balance = await stakeContract
+        .connect(alice)
+        .balanceOf(alice.address)
+      expect(balance).to.equal('495')
+    })
+
+    it('withdrawAndStake reverts with wrong token', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transferFrom.returns(true)
```
Comment (AboldUSER) [inline source/pool/test/unit.js:380]: Present from previous commit, though also not necessary
```
await feeToken.mock.balanceOf.returns('100000')
await feeToken.mock.transfer.returns(true)
```

File: source/pool/test/unit.js [hunk 5]
```diff
@@ -1,607 +1,653 @@
...
+      expect(isClaimed).to.equal(true)
+
+      const balance = await stakeContract.connect(bob).balanceOf(bob.address)
+      expect(balance).to.equal('495')
+    })
+
+    it('withdrawAndStakeFor reverts with wrong token', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
+      await feeToken.mock.transferFrom.returns(true)
```
Comment (AboldUSER) [inline source/pool/test/unit.js:437]: Present from previous commit, though also not necessary
```
await feeToken.mock.balanceOf.returns('100000')
await feeToken.mock.transfer.returns(true)
await feeToken.mock.transferFrom.returns(true)
```

File: source/pool/test/unit.js [hunk 6]
```diff
@@ -1,607 +1,653 @@
...
+
+    it('Test addAdmin reverts', async () => {
+      await expect(
+        pool.connect(alice).addAdmin(alice.address)
+      ).to.be.revertedWith('Ownable: caller is not the owner')
+    })
+
+    it('Test addAdmin reverts with zero address', async () => {
+      await expect(
+        pool.connect(deployer).addAdmin('0x0000000000000000000000000000000000000000')
```
Comment (AboldUSER) [inline source/pool/test/unit.js:537]: Same zero address comment as above

File: source/pool/test/unit.js [hunk 7]
```diff
@@ -1,607 +1,653 @@
...
+      await expect(await pool.connect(alice).enable(root)).to.emit(
+        pool,
+        'Enable'
+      )
+      expect(await pool.roots(root)).to.equal(true)
+    })
+
+    it('Test setclaimed with admin is successful', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
```
Comment (AboldUSER) [inline source/pool/test/unit.js:572]: Present from previous commit, though also not necessary
```
await feeToken.mock.balanceOf.returns('100000')
await feeToken.mock.transfer.returns(true)
```

File: source/pool/test/unit.js [hunk 8]
```diff
@@ -1,607 +1,653 @@
...
+            },
+          ],
+          feeToken.address
+        )
+      ).to.be.revertedWith('CLAIM_ALREADY_MADE')
+    })
+
+    it('Test setclaimed with non-admin reverts', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
```
Comment (AboldUSER) [inline source/pool/test/unit.js:598]: These lines are not necessary:
```
await feeToken.mock.balanceOf.returns('100000')
await feeToken.mock.transfer.returns(true)
```

File: source/pool/test/unit.js [hunk 9]
```diff
@@ -1,607 +1,653 @@
...
+        .setClaimed(
+          root,
+          [bob.address]
+        )
+      ).to.be.revertedWith('NOT_ADMIN')
+    })
+
+    it('Test setclaimed reverts with claim already made', async () => {
+      await feeToken.mock.balanceOf.returns('100000')
+      await feeToken.mock.transfer.returns(true)
```
Comment (AboldUSER) [inline source/pool/test/unit.js:614]: These lines are not necessary:
```
await feeToken.mock.balanceOf.returns('100000')
await feeToken.mock.transfer.returns(true)
```

**Review by AboldUSER — CHANGES_REQUESTED** [review]

**Comment by francisldn** [comment]
Thank you @AboldUSER. Agree with your comments and have amended and tested them accordingly.

**Review by AboldUSER — APPROVED** [review]

**Pull request merged**